instrument_config/mwa_dipole_mutual_coupling.pro

```diff
@@ -4,19 +4,17 @@ file_path_Z_matrix=filepath('mwa_ZMatrix.fits',root=rootdir('FHD'),sub='instrume
 file_path_Z_LNA=filepath('mwa_LNA_impedance.sav',root=rootdir('FHD'),sub='instrument_config')
 n_dipole=16
 n_ant_pol=2
-icomp=Complex(0,1)
+icomp=DComplex(0,1)
 
 status=0
 fits_info,file_path_Z_matrix,/silent,n_ext=n_ext
 n_ext+=1 ;n_ext starts counting AFTER the 0th extension, which it considers to be the main data unit, but we use that one too
-Zmat_arr=Complexarr(n_ext,n_ant_pol,n_dipole,n_dipole)
-;Zmat_arr_X=Complexarr(n_ext,n_dipole,n_dipole)
-;Zmat_arr_Y=Complexarr(n_ext,n_dipole,n_dipole)
+Zmat_arr=DComplexarr(n_ext,n_ant_pol,n_dipole,n_dipole)
 freq_arr_Zmat=Fltarr(n_ext)
 FOR ext_i=0,n_ext-1 DO BEGIN
     Zmat1=mrdfits(file_path_Z_matrix,ext_i,header,status=status,/silent)
     Zmat=Zmat1[*,*,0]*(Cos(Zmat1[*,*,1])+icomp*Sin(Zmat1[*,*,1]))
-    freq_arr_Zmat[ext_i]=Float(sxpar(header,'FREQ'))
+    freq_arr_Zmat[ext_i]=Double(sxpar(header,'FREQ'))
```

Comment:
`freq_arr_Zmat` should probably be left as floats, just because that's what the value read from the header will be. If you anticipate that changing at some point in the future, then you should also change its definition on line 12 above to be a double.

Reply:
The value is read as a double for the frequencies. I've changed the initialisation of the array to be doubles to be more concise.